src/me/mrCookieSlime/Slimefun/listeners/ArmorListener.java

```diff
@@ -32,48 +34,73 @@ public ArmorListener(SlimefunPlugin plugin) {
 		plugin.getServer().getPluginManager().registerEvents(this, plugin);
 	}
 	
-	@EventHandler(priority=EventPriority.MONITOR)
+	@EventHandler(priority = EventPriority.MONITOR)
 	public void onDamage(EntityDamageEvent e) {
 		if (e.getEntity() instanceof Player && !e.isCancelled()) {
 			Player p = (Player) e.getEntity();
 			for (ItemStack armor: p.getInventory().getArmorContents()) {
-				if (armor != null && SlimefunItem.getByItem(armor) != null) {
-					if (SlimefunItem.getByItem(armor).isItem(SlimefunItems.ENDER_BOOTS) && Slimefun.hasUnlocked(p, SlimefunItems.ENDER_BOOTS, true)) {
-						if (e instanceof EntityDamageByEntityEvent && ((EntityDamageByEntityEvent) e).getDamager() instanceof EnderPearl) e.setCancelled(true);
+				SlimefunItem item = SlimefunItem.getByItem(armor);
+				if (item != null) {
+					if (item.getID().equals("ENDER_BOOTS")
+						&& Slimefun.hasUnlocked(p, SlimefunItems.ENDER_BOOTS, true)
```

Comment:
Sorry, that I didn't point this out earlier.

Slimefun.hasUnlocked() makes a call to Slimefunitem.getByItem() that is not neccessary here.
Instead you can just use Slimefun.hasUnlocked(Player, SlimefunItem, boolean) instead and directly pass the item variable.

That'd increase performance a lot as well.
This also applies to all the other occasions down below.